lib/insights_cloud/async/connector_playbook_execution_reporter_task.rb

```diff
@@ -0,0 +1,192 @@
+        correlation_id = invocation_inputs['correlation_id']
+
+        plan_self(
+          report_url: report_url,
+          report_interval: report_interval,
+          job_invocation_id: job_invocation.id,
+          correlation_id: correlation_id
+        )
+      end
+
+      def run(event = nil)
+        # Handle skip events
+        return if event == Dynflow::Action::Skip
+
+        super
+      end
+
+      def rescue_strategy_for_self
+        Dynflow::Action::Rescue::Skip
+      end
+
+      def done?(current_status = invocation_status)
+        job_invocation.finished? || current_status.map { |_host_id, task_status| task_status['report_done'] }.all?
+      end
+
+      # noop, we don't want to do anything when the polling task starts
+      def invoke_external_task
+        poll_external_task
+      end
+
+      def poll_external_task
+        current_status = invocation_status
+        report_job_progress(current_status)
+        # record the current state and increment the sequence for the next invocation
+        {
+          invocation_status: current_status,
+        }
+      end
+
+      def poll_intervals
+        [report_interval]
+      end
+
+      private
+
+      def connector_playbook_job?(job_invocation)
+        job_invocation&.remote_execution_feature_id == connector_feature_id
+      end
+
+      def connector_feature_id
+        self.class.connector_feature_id
+      end
+
+      def invocation_inputs(job_invocation)
+        Hash[
+          job_invocation.pattern_template_invocations.first.input_values.map do |input_value|
+            [input_value.template_input.name, input_value.value]
+          end
+        ]
+      end
+
+      def job_invocation
+        @job_invocation ||= JobInvocation.find(input['job_invocation_id'])
+      end
+
+      def report_interval
+        @report_interval ||= input['report_interval']
+      end
+
+      def correlation_id
+        @correlation_id ||= input['correlation_id']
+      end
+
+      def host_status(host)
+        external_task&.dig('invocation_status', host)
+      end
+
+      def sequence(host)
+        host_status(host)&.fetch('sequence', 0).to_i
+      end
+
+      def host_done?(host)
+        ActiveModel::Type::Boolean.new.cast(host_status(host)&.fetch('report_done', nil))
+      end
+
+      def report_url
+        input['report_url']
+      end
+
+      def invocation_status
+        Hash[job_invocation.targeting.hosts.map do |host|
+          next unless host.insights&.uuid
+          [
+            host.insights.uuid,
+            task_status(job_invocation.sub_task_for_host(host), host.insights.uuid),
+          ]
+        end.compact]
+      end
+
+      def task_status(host_task, host_name)
+        unless host_task
+          return { 'state' => 'unknown' }
+        end
+
+        {
+          'state' => host_task.state,
+          'output' => host_task.main_action.live_output.map { |line| line['output'] }.join("\n"),
```

Comment:
To be honest I'm not sure if the live output is sorted by default or not

Reply:
I think it is: https://github.com/theforeman/foreman_remote_execution/blob/0292f908250bb10275138e021e879267885d220d/app/lib/actions/remote_execution/run_host_job.rb#L105